incubator/binding-pgsql/src/main/java/io/aklivity/zilla/runtime/binding/pgsql/internal/stream/PgsqlServerFactory.java

```diff
@@ -0,0 +1,1353 @@
+import io.aklivity.zilla.runtime.binding.pgsql.internal.types.stream.WindowFW;
+import io.aklivity.zilla.runtime.engine.EngineContext;
+import io.aklivity.zilla.runtime.engine.binding.BindingHandler;
+import io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer;
+import io.aklivity.zilla.runtime.engine.buffer.BufferPool;
+import io.aklivity.zilla.runtime.engine.config.BindingConfig;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlCompletedFlushExFW;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlDataExFW;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlFlushExFW;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlQueryDataExFW;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlReadyFlushExFW;
+import io.aklivity.zilla.specs.binding.pgsql.internal.types.stream.PgsqlTypeFlushExFW;
+
+public final class PgsqlServerFactory implements PgsqlStreamFactory
+{
+    private static final DirectBuffer EMPTY_BUFFER = new UnsafeBuffer(new byte[0]);
+    private static final OctetsFW EMPTY_OCTETS = new OctetsFW().wrap(EMPTY_BUFFER, 0, 0);
+    private static final Consumer<OctetsFW.Builder> EMPTY_EXTENSION = ex -> {};
+
+    private static final int FLAGS_INIT = 0x02;
+    private static final int FLAGS_CONT = 0x00;
+    private static final int FLAGS_FIN = 0x01;
+    private static final int FLAGS_COMP = 0x03;
+
+    private final MutableInteger payloadRemaining = new MutableInteger(0);
+
+    private final BeginFW beginRO = new BeginFW();
+    private final DataFW dataRO = new DataFW();
+    private final EndFW endRO = new EndFW();
+    private final AbortFW abortRO = new AbortFW();
+    private final FlushFW flushRO = new FlushFW();
+
+    private final BeginFW.Builder beginRW = new BeginFW.Builder();
+    private final DataFW.Builder dataRW = new DataFW.Builder();
+    private final EndFW.Builder endRW = new EndFW.Builder();
+    private final AbortFW.Builder abortRW = new AbortFW.Builder();
+    private final FlushFW.Builder flushRW = new FlushFW.Builder();
+
+    private final ResetFW resetRO = new ResetFW();
+    private final WindowFW windowRO = new WindowFW();
+
+    private final ResetFW.Builder resetRW = new ResetFW.Builder();
+    private final WindowFW.Builder windowRW = new WindowFW.Builder();
+
+    private final ExtensionFW extensionRO = new ExtensionFW();
+    private final PgsqlDataExFW pgsqlDataExRO = new PgsqlDataExFW();
+    private final PgsqlFlushExFW pgsqlFlushExRO = new PgsqlFlushExFW();
+
+    private final PgsqlDataExFW.Builder dataExRW = new PgsqlDataExFW.Builder();
+
+
+    private final PgsqlMessageFW messageRO = new PgsqlMessageFW();
+
+    private final PgsqlMessageFW.Builder messageRW = new PgsqlMessageFW.Builder();
+
+    private final BufferPool bufferPool;
+    private final MutableDirectBuffer writeBuffer;
+    private final MutableDirectBuffer frameBuffer;
```

Comment:
In the PGSQL protocol specification, they use the terminology "message", not "frame".

Let's replace all references to "frame" with "message" here to convey the same concept precisely.